slidez-core/src/main/java/com/binarystudio/academy/slidez/domain/user/model/User.java

```diff
@@ -29,24 +26,50 @@
 	@Column(name = "id", updatable = false, nullable = false)
 	private UUID id;
 
-	@Column(name = "email", unique = true)
+	@Column(name = "email", unique = true, nullable = false)
 	private String email;
 
-	@Column
-	private String firstName;
-
-	@Column
-	private String lastName;
-
-	@Column
+	@Column(name = "password", nullable = false)
 	private String password;
 
-	@Column(unique = true)
+	@Column(name = "role", nullable = false)
 	@Enumerated(EnumType.STRING)
 	private UserRole role;
 
-	public boolean isAdmin() {
-		return role == UserRole.ADMIN;
+	@OneToOne(mappedBy = "user")
+	private UserProfile userProfile;
+
+	@OneToOne(mappedBy = "user")
+	private GoogleCredentials googleCredentials;
+
+	@OneToMany(fetch = FetchType.LAZY, cascade = CascadeType.ALL, orphanRemoval = true)
```

Comment:
I think we should use EAGER fetching here.

Reply:
We should not use eager on user. We will often reject email+password combination so fetching other data will just create useless joins. If I remember correctly, in addon we will need only email and password